File: short program 3/Vending_machine.py
def GetOption(Sweet):
    return Sweet.option
def GetName(Sweet):
    return Sweet.name
def GetPrice(Sweet):
    return Sweet.price
def AskForOption(Sweets_array):
    choice = raw_input("Choose from\nA: Packet of Crisps\nB: Health Bar\nC: Bar of Chocolate\nOption: ")
    found = False
    for i in range(3):
        if GetOption(Sweets_array[i])==choice:
            index = i
            found = True
    while found==False:
        print "Please enter valid option"
        AskForOption(Sweets_array)
    return index
def PopArray(Sweets, Sweets_array):
    Sweet = Sweets("A", "A Packet of Crisps", 1.50)
    Sweets_array.append(Sweet)
    Sweet = Sweets("B", "A Health Bar", 1.20)
    Sweets_array.append(Sweet)
    Sweet = Sweets("C", "A Bar of Chocolate", 2.00)
    Sweets_array.append(Sweet)
    return Sweets_array
def PrintResponse(name, price):
    print name, " cost is ", price
def main():
    Sweets_array=[]
    Sweets_array = PopArray(Sweets, Sweets_array)
    index = AskForOption(Sweets_array)
    name = GetName(Sweets_array[index])
    price = GetPrice(Sweets_array[index])
    PrintResponse(name, price)
class Sweets:
    def __init__(self, option, name, price):
        self.option = option
        self.name = name
        self.price = price
main()

File: short program 3/Vending_machine.java
//Author: Rohan Dhother
//Date: 10/10/2019
import java.util.Scanner;
class Vending_machine
{
	public static void main (String[] param) //This is the main method of the class
	{
		AskForOption();
		System.exit(0);
	}
	public static void AskForOption() //This method asks the user for input of which a option, then prints a response
	{
		Scanner scanner = new Scanner(System.in);
		System.out.println("Choose from (A: Packet of Crisps, B: Health Bar, C: Bar of Chocolate)? ");
		String ans = scanner.nextLine();
		if (ans.equals("A"))
		{
			System.out.println("A Packet of Crisps costs 1.50");
		}
		else if (ans.equals("B"))
		{
			System.out.println("A Health Bar costs 1.20");
		}
		else if (ans.equals("C"))
		{
			System.out.println("A Bar of Chocolate costs 2.00");
		}
		else
		{
			System.out.println("I do not have anything for choice " + ans);
		}
	}
}
